R/supressCount.R

```diff
@@ -52,52 +46,36 @@ supressCount <- function(result,
       "- `cohort_definition_id` is not a column of result"
     )
   }
-  column2Check <- c("variable") %in% colnames(result)
-  if (!isTRUE(column2Check)) {
-    errorMessage$push(
-      "- `variable` is not a column of result"
-    )
-  }
-  column3Check <- c("estimate") %in% colnames(result)
-  if (!isTRUE(column3Check)) {
-    errorMessage$push(
-      "- `estimate` is not a column of result"
-    )
-  }
-  column4Check <- c("value") %in% colnames(result)
-  if (!isTRUE(column4Check)) {
-    errorMessage$push(
-      "- `value` is not a column of result"
-    )
-  }
-  checkmate::assertIntegerish(minimumCellCounts, len = 1,
-                              add = errorMessage,
-  )
-  checkmate::assertCharacter(globalVariables,
-                             add = errorMessage,
+
+  checkmate::assertIntegerish(minimumCellCount,
+    len = 1,
+    add = errorMessage,
   )
   checkmate::assertCharacter(estimatesToObscure,
-                             add = errorMessage,
+    add = errorMessage,
   )
   checkmate::reportAssertions(collection = errorMessage)
 
   # Start code
-  values_to_osbcure <- suppressWarnings(as.numeric(result$value)) <
-    minimumCellCounts &
-    suppressWarnings(as.numeric(result$value)) > 0
-  obscured_values <- result$estimate %in% estimatesToObscure & values_to_osbcure
-  obscured_cohort <- unique(result$cohort_definition_id[
-    result$estimate %in% estimatesToObscure &
-      result$variable %in% globalVariables &
-      values_to_osbcure
-  ])
-  result$value[obscured_values] <- paste0("<", minimumCellCounts)
-  result$value[
-    result$cohort_definition_id %in% obscured_cohort
-  ] <- as.character(NA)
-  result$value[
-    result$cohort_definition_id %in% obscured_cohort &
-      result$variable %in% globalVariables
-  ] <- paste0("<", minimumCellCounts)
+  if (!is.null(cohort_definition_id)) {
+    result_to_obscure <- result %>%
+      dplyr::filter(cohort_definition_id %in% .env$cohort_definition_id)
+  }else{result_to_obscure <- result}
+
+  result_to_obscure <- result_to_obscure %>% dplyr::rename(estimatesToObscure := !!estimatesToObscure)
```

Comment:
why is this necessary?

Reply:
> why is this necessary?

if they only want to suppress one cohort definition id?